uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-k8s/src/main/java/org/uberfire/java/nio/fs/k8s/K8SFileSystemProvider.java

```diff
@@ -0,0 +1,337 @@
+                                                                        K8SBasicFileAttributeView.class,
+                                                                        options);
+            return view.readAttributes();
+        }
+        return null;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected <V extends FileAttributeView> V createFileAttributeView(final GeneralPathImpl path, 
+                                                                      final Class<V> type) {
+        if (AbstractBasicFileAttributeView.class.isAssignableFrom(type)) {
+            final V newView = (V) new K8SBasicFileAttributeView(path, this);
+            path.addAttrView(newView);
+            return newView;
+        } else {
+            return null;
+        }
+    }
+    
+    @Override
+    public void copy(final Path sourceIn,
+                     final Path targetIn,
+                     final CopyOption... options) {
+        checkNotNull("source", sourceIn);
+        checkNotNull("target", targetIn);
+        Path source = toAbsoluteRealPath(sourceIn);
+        Path target = toAbsoluteRealPath(targetIn);
+
+        Optional<ConfigMap> srcCMOpt = executeCloudFunction(
+            client -> getFsObjCM(client, source), KubernetesClient.class);
+        checkCondition("source must exist", srcCMOpt.isPresent());
+        checkFileExistsThenThrow(target);
+
+        ConfigMap srcCM = srcCMOpt.orElseThrow(IllegalArgumentException::new);
+        if (isDirectory(srcCM)) {
+            throw new UnsupportedOperationException(srcCM.getMetadata().getName() + "is a directory.");
+        }
+        
+        String content = srcCM.getData().getOrDefault(CFG_MAP_FSOBJ_CONTENT_KEY, "");
+        long size = Long.parseLong(srcCM.getMetadata().getAnnotations().getOrDefault(CFG_MAP_ANNOTATION_FSOBJ_SIZE_KEY, "0"));
+        executeCloudFunction(client -> createOrReplaceFSCM(client, 
+                                                           target,
+                                                           createOrReplaceParentDirFSCM(client, target, size, false),
+                                                           Collections.singletonMap(CFG_MAP_FSOBJ_CONTENT_KEY, content),
+                                                           false), 
+                             KubernetesClient.class);
+    }
+
+    @Override
+    public void move(final Path sourceIn,
+                     final Path targetIn,
+                     final CopyOption... options) {
+        Path source = toAbsoluteRealPath(sourceIn);
+        Path target = toAbsoluteRealPath(targetIn);
+        try {
+            copy(source, target);
+        } catch (Exception e) {
+            try {
+                delete(target);
+            } catch (NoSuchFileException nsfe) {
+                throw new IOException("Moving file failed due to Copy Source Exception [" + e.getMessage() + "].");
```

Comment:
Doesn't belong this to the first try catch block?

Reply:
It guards the situation which error happens before creating the target file. In such case, the 'delete' method will fail with 'NoSuchFileException'.